src/ring_theory/ideal/basic.lean

```diff
@@ -269,6 +269,14 @@ end
 theorem exists_maximal [nontrivial α] : ∃ M : ideal α, M.is_maximal :=
 let ⟨I, ⟨hI, _⟩⟩ := exists_le_maximal (⊥ : ideal α) submodule.bot_ne_top in ⟨I, hI⟩
 
+lemma ideals_nontrivial_of_nontrivial [nontrivial α] : nontrivial (ideal α) :=
```

Comment:
Is there a reason to not make this an instance?
```suggestion
instance [nontrivial α] : nontrivial (ideal α) :=
```

Reply:
Actually, thinking about this: for any nontrivial (semi)module `M` we have `submodule R M` is nontrivial. Namely, we have `x ≠ 0 : M` in `\top` and not in `\bot`. And it doesn't look like we have that instance yet. Could you do the generalization?

Reply:
I will certainly! In exchange, would you explain what is an `instance` for Lean and how it treats an `instance` differently from a `lemma`? :P